UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Services/ProductDataServiceTests.cs

```diff
@@ -0,0 +1,73 @@
+﻿using FakeItEasy;
+using FluentValidation.Results;
+using NUnit.Framework;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.API.Services;
+using UKHO.ExchangeSetService.API.Validation;
+using UKHO.ExchangeSetService.Common.Models.Request;
+using UKHO.ExchangeSetService.Common.Models.Response;
+
+namespace UKHO.ExchangeSetService.API.UnitTests.Services
+{
+    [TestFixture()]
+    public class ProductDataServiceTests
+    {
+        private IProductDataSinceDateTimeValidator fakeIProductDataValidatorService;
+        private ProductDataService service;
+
+        [SetUp]
+        public void Setup()
+        {
+            fakeIProductDataValidatorService = A.Fake<IProductDataSinceDateTimeValidator>();
+            service = new ProductDataService(fakeIProductDataValidatorService);
+        }
+
+        [Test()]
+        public async Task WhenInvalidSinceDateTimeInRequest_ThenValidateProductDataSinceDateTimeReturnsBadRequest()
+        {
+            A.CallTo(() => fakeIProductDataValidatorService.Validate(A<ProductDataSinceDateTimeRequest>.Ignored))
+                .Returns(new ValidationResult(new List<ValidationFailure>
+                            { new ValidationFailure("SinceDateTime", "Provided since date time is either invalid or invalid format, the valid format is 'RFC1123 format'.")}));
+
+            var result = await service.ValidateProductDataSinceDateTime(new ProductDataSinceDateTimeRequest());
+            Assert.IsFalse(result.IsValid);
+            Assert.AreEqual("Provided since date time is either invalid or invalid format, the valid format is 'RFC1123 format'.", result.Errors.Single().ErrorMessage);
+        }
+
+        [Test()]
+        public async Task WhenSinceDateTimeFormatIsGreaterThanCurrrentDateTimeInRequest_ThenValidateProductDataSinceDateTimeReturnsBadRequest()
+        {
+            A.CallTo(() => fakeIProductDataValidatorService.Validate(A<ProductDataSinceDateTimeRequest>.Ignored))
+                .Returns(new ValidationResult(new List<ValidationFailure>
+                            { new ValidationFailure("SinceDateTime", "Provided since date time cannot be a future date.")}));
+
+            var result = await service.ValidateProductDataSinceDateTime(new ProductDataSinceDateTimeRequest());
+            Assert.IsFalse(result.IsValid);
+            Assert.AreEqual("Provided since date time cannot be a future date.", result.Errors.Single().ErrorMessage);
+        }
+
+        [Test()]
+        public async Task WhenCallbackUrlparameterNotValidInRequest_ThenValidateProductDataSinceDateTimeReturnsBadRequest()
```

Comment:
```suggestion
        public async Task WhenCallbackUrlParameterNotValidInRequest_ThenValidateProductDataSinceDateTimeReturnsBadRequest()
```

Reply:
We have change method naming convention.